pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -169,65 +167,58 @@ func getPullRequests(
 	ghs *githubScraper,
 	ctx context.Context,
 	client graphql.Client,
-	repos []SearchNodeRepository,
+	repo SearchNodeRepository,
 	now pcommon.Timestamp,
-	pullRequestCh chan []PullRequestNode,
-	waitGroup *sync.WaitGroup,
-) {
-	defer waitGroup.Done()
-
-	for _, repo := range repos {
-		var repoName string = repo.Name
-
-		//var defaultBranch string
+) ([]PullRequestNode, error) {
 
-		prPages, err := getNumPrPages(ghs, ctx, client, repoName, now)
-		if err != nil {
-			ghs.logger.Sugar().Errorf("error getting total pr pages", zap.Error(err))
-		}
-		pullRequests, err := getPrData(ghs, ctx, client, prPages, repoName, ghs.cfg.GitHubOrg)
-		if err != nil {
-			ghs.logger.Sugar().Errorf("error getting pr data", zap.Error(err))
-		}
+	var repoName string = repo.Name
+	prPages, err := getNumPrPages(ghs, ctx, client, repoName, now)
+	if err != nil {
+		ghs.logger.Sugar().Errorf("error getting total pr pages", zap.Error(err))
+		return nil, err
+	}
+	pullRequests, err := getPrData(ghs, ctx, client, prPages, repoName, ghs.cfg.GitHubOrg)
+	if err != nil {
+		ghs.logger.Sugar().Errorf("error getting pr data", zap.Error(err))
+		return nil, err
+	}
 
-		if pullRequests != nil {
-			pullRequestCh <- pullRequests
-		}
+	if pullRequests != nil {
+		return pullRequests, nil
 	}
+	return nil, nil
 }
 
 func processPullRequests(
 	ghs *githubScraper,
 	ctx context.Context,
 	client graphql.Client,
 	now pcommon.Timestamp,
-	pullRequests chan []PullRequestNode,
+	pullRequests []PullRequestNode,
 ) {
-
-	for prs := range pullRequests {
-		for _, pr := range prs {
-			repoName := pr.Repository.Name
-			if pr.Merged {
-				prMergedTime := pr.MergedAt
-				mergeAge := int64(prMergedTime.Sub(pr.CreatedAt).Hours())
-				ghs.mb.RecordGitRepositoryPullRequestMergeTimeDataPoint(now, mergeAge, repoName, pr.HeadRefName)
-				//only exists if the pr is merged
-				if pr.MergeCommit.Deployments.TotalCount > 0 {
-					deploymentAgeUpperBound := pr.MergeCommit.Deployments.Nodes[0].CreatedAt
-					deploymentAge := int64(deploymentAgeUpperBound.Sub(pr.CreatedAt).Hours())
-					ghs.mb.RecordGitRepositoryPullRequestDeploymentTimeDataPoint(now, deploymentAge, repoName, pr.HeadRefName)
-				}
-			} else {
-				prAge := int64(now.AsTime().Sub(pr.CreatedAt).Hours())
-				ghs.mb.RecordGitRepositoryPullRequestTimeDataPoint(now, prAge, repoName, pr.HeadRefName)
-			}
-			if pr.Reviews.TotalCount > 0 {
-				approvalAgeUpperBound := pr.Reviews.Nodes[0].CreatedAt
-				approvalAge := int64(approvalAgeUpperBound.Sub(pr.CreatedAt).Hours())
-				ghs.mb.RecordGitRepositoryPullRequestApprovalTimeDataPoint(now, approvalAge, repoName, pr.HeadRefName)
+	for _, pr := range pullRequests {
+		repoName := pr.Repository.Name
+		if pr.Merged {
```

Comment:
This set of conditionals could be reversed to reduce code & not need the else statement

Reply:
Wouldn't that be the same thing since its either one or the other regardless of the order?  If the else statement wasn't there then you'd be doing both sometimes.